fabric-api-base/src/main/java/net/fabricmc/fabric/api/util/TriState.java

```diff
@@ -16,16 +16,124 @@
+ */
 public enum TriState {
+	/**
+	 * Represents the boolean value of {@code false}.
+	 */
 	FALSE,
+	/**
+	 * Represents a value that refers to a "default" value, often as a fallback.
+	 */
 	DEFAULT,
+	/**
+	 * Represents the boolean value of {@code true}.
+	 */
 	TRUE;
 
-	public static TriState of(boolean b) {
-		return b ? TRUE : FALSE;
+	/**
+	 * Gets the corresponding tri-state from a boolean value.
+	 *
+	 * @param bool the boolean value
+	 * @return {@link TriState#TRUE} or {@link TriState#FALSE} depending on the value of the boolean.
+	 */
+	public static TriState of(boolean bool) {
+		return bool ? TRUE : FALSE;
+	}
+
+	/**
+	 * Gets a tri-state from a nullable boxed boolean.
+	 *
+	 * @param bool the boolean value
+	 * @return {@link TriState#DEFAULT} if {@code null}.
+	 * Otherwise {@link TriState#TRUE} or {@link TriState#FALSE} depending on the value of the boolean.
+	 */
+	public static TriState of(@Nullable Boolean bool) {
+		return bool == null ? DEFAULT : of(bool.booleanValue());
 	}
 
+	/**
+	 * Gets the value of the tri-state.
+	 *
+	 * @return true if the tri-state is {@link TriState#TRUE}.
+	 * Otherwise false.
+	 */
 	public boolean get() {
 		return this == TRUE;
 	}
+
+	/**
+	 * Gets the value of the tri-state as a boxed, nullable boolean.
+	 *
+	 * @return {@code null} if {@link TriState#DEFAULT}.
+	 * Otherwise {@code true} if {@link TriState#TRUE} or {@code false} if {@link TriState#FALSE}.
+	 */
+	@Nullable
+	public Boolean getBoxed() {
+		return this == DEFAULT ? null : this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param value the value to fallback to
+	 * @return the value of the tri-state or the supplied value if {@link TriState#DEFAULT}.
+	 */
+	public boolean orElse(boolean value) {
+		return this == DEFAULT ? value : this.get();
+	}
+
+	/**
+	 * Gets the value of this tri-state.
+	 * If the value is {@link TriState#DEFAULT} then use the supplied value.
+	 *
+	 * @param supplier the supplier used to get the value to fallback to
+	 * @return the value of the tri-state or the value of the supplier if the tri-state is {@link TriState#DEFAULT}.
+	 */
+	public boolean orElseGet(BooleanSupplier supplier) {
+		return this == DEFAULT ? supplier.getAsBoolean() : this.get();
+	}
+
+	/**
+	 * Maps the boolean value of this tri-state if it is {@link TriState#TRUE} or {@link TriState#FALSE}
+	 *
+	 * @param mapper the mapper to use
+	 * @param <T> the type of object being supplier by the mapper
+	 * @return an optional containing the mapped value; {@link Optional#empty()} if the tri-state is {@link TriState#DEFAULT} or the value provided by the mapper is {@code null}.
+	 */
+	public <T> Optional<T> map(BooleanFunction<@Nullable ? extends T> mapper) {
+		Objects.requireNonNull(mapper, "Mapper function cannot be null");
+
+		if (this.getBoxed() == null) {
```

Comment:
everything else checks `this == DEFAULT` instead, should probably be changed here, too, for consistency.